### PR TITLE
Fix integration workflows to run if needed

### PR DESCRIPTION
**Pull request by niklassiemer**
I hope I finally fixed all the bugs on when the workflows are executed! Some testing in https://github.com/niklassiemer/test_checkout/pull/6

Compatibility workflows correctly
- [x]  skipped if no integration label (a67d5db)
- [x] started on setting integration label
- [x] started on pushing to the PR if integration label is there (c6adc1f)
- [x] skipped on pushing to the PR if the integration label was removed (023aa26)
- [x] started on reopen of the PR if the integration label is there (2fb2f72)
- [x] started on merge with master, although the integration label was removed (6640f11)

### CODE REVIEW

File: .github/workflows/atomistics-compat.yml [hunk 1]
```diff
@@ -13,10 +13,9 @@ on:
 jobs:
   build:
     if: |
-      ${{ github.event.type == 'PushEvent' ||
-          ( github.event.type == 'PullRequestEvent'  && contains(github.event.pull_request.labels.*.name, 'integration' ))
-       }}
-
+      github.event_name == 'push' ||
```
Comment (max-hassani) [inline .github/workflows/atomistics-compat.yml:16]: Doesn't this `or` condition cause the integration tests to run on all the pushes to the master branch?

Comment (niklassiemer) [reply .github/workflows/atomistics-compat.yml:16]: Yes, and I think this should be the intended behavior. In general, we only push the merge commit to the master branch. And every time the master is changed I would run the full test suite with all integration tests to validate the new state. If someone did not use the `integration` label and a merge to the master is failing on the integration tests we are directly alerted.

**Comment by niklassiemer** [comment]
Last test missing is the actual push to master - I think everything is finally fine and working as intended! I will add an integration workflow to the https://github.com/pyiron/pyiron_module_template to avoid future headaches like this.

**Review by max-hassani — APPROVED** [review]
LGTM! Thanks for the verifying that the test work as intended in the check boxes :)

**Comment by niklassiemer** [comment]
Well then I merge and let's hope the last box gets a check 😅

**Comment by niklassiemer** [comment]
🎉 they started 😃

**Pull request merged**